### PR TITLE
feat: Added "Scale dataplane to zero" workflow to the documentation

### PR DESCRIPTION
**Pull request by jparrill**
Documented the "Scale data plane to zero" workflow to allow the customers to save money and compute time. It's mainly focussed in AWS but should be provider agnostic.

Signed-off-by: Juan Manuel Parrilla Madrid <jparrill@redhat.com>

**Which issue(s) this PR fixes** 
Fixes #[HOSTEDCP-576](https://issues.redhat.com//browse/HOSTEDCP-576)

**Checklist**
- [X] Subject and description added to both, commit and PR.
- [X] Relevant issues have been referenced.
- [X] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: docs/content/how-to/hibernate-workers.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
+title: Hibernating Hosted Control Plane Worker nodes
```
Comment (zanetworker) [inline docs/content/how-to/hibernate-workers.md:2]: Maybe we call it scaling cluster to zero nodes. Hibernation could mean more than just scaling and has more nuance (e.g., shutdown vs remove nodes).

File: docs/content/how-to/hibernate-workers.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+- We will add an annotate to the nodes which will ensure the pod drainning does not happen. This we will save time and money and also we will avoid stucked pods.
+
+Now let's explain the workflow.
+
+## Workflow
+
+This is a not difficult procedure but you need to be sure that you have set the right `KUBECONFIG` and you point to the correct context.
+
+- You need to work over the Management Cluster (The one where you have installed Hypershift)
+- Now you need to annotate the `Machine` object from the Hosted Control Plane namespace with "machine.cluster.x-k8s.io/exclude-node-draining="
```
Comment (zanetworker) [inline docs/content/how-to/hibernate-workers.md:26]: Why is this needed? can we document that?

File: docs/content/how-to/hibernate-workers.md (hunk 1 above)
Comment (enxebre) [reply docs/content/how-to/hibernate-workers.md:2]: Yes, let's don't call use the term hibernation in this doc.
This is just a SOP that proves how to temporarily shrink to zero your data plane.

File: docs/content/how-to/hibernate-workers.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+- **This is a destructive action**, all the workloads in the worker nodes will dissapear.
+- The Hosted Control Plane will stay up and running, and you can scale up the node pool whenever you want.
+- Some pods in the control plane will stay in "Pending" state.  
+- Once you rescale the NodePool/s it will take time until they reach the fully **Ready** state.
+- We will add an annotate to the nodes which will ensure the pod drainning does not happen. This we will save time and money and also we will avoid stucked pods.
+
+Now let's explain the workflow.
+
+## Workflow
```
Comment (enxebre) [inline docs/content/how-to/hibernate-workers.md:21]: Let's elaborate this section presenting our semantics and limitations. E.g:
"To temporarily remove you data plane compute all you need to do is scaling all NodePools down to zero. By default draining will be honoured to protect any workloads. You can choose to deliberately skip draining by setting [drainTimeout](https://github.com/openshift/hypershift/blob/main/api/v1alpha1/nodepool_types.go#L130-L137) to a lower value - Caveat: changing drainTimeout in an existing NodePool will trigger a rolling update first and so the new value will only affect to the new created Machines. If you are in a situation where the intent is skip draining and don't want to assume the rolling upgrade triggered by changing the field, you can skip draining forcefully in a given Machine by setting "machine.cluster.x-k8s.io/exclude-node-draining=". This is a known limitation and we plane to prevent changing drainTimeout from triggering a rolling update in the future. Machines.machine.cluster.x-k8s.io are considered a lower level resource and any consumer interaction is recommended to happen via NodePool.

File: docs/content/how-to/hibernate-workers.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    echo "HC ${3} does not exists in the namespace ${2} of the Management Cluster"
+    exit 1
+fi
+
+scale_down_nodepool
+```
+
+</details>
+
+After these steps, you will see how the (in the AWS case) instances will be terminated instantly, but Openshift will take some time until the nodes get deleted because of the default timeouts set on the platforms.
```
Comment (enxebre) [inline docs/content/how-to/hibernate-workers.md:120]: I'm not sure what this means. The Node will be deleted by CAPI as soon and only when the underlying instance has gone away. -  [This is to prevent issues from happening with e.g. stateful](https://github.com/kubernetes-sigs/cluster-api/issues/2565)

File: docs/mkdocs.yml [hunk 5]
```diff
@@ -43,6 +43,7 @@ nav:
   - how-to/external-dns.md
   - how-to/metrics-sets.md
   - how-to/node-tuning.md
+  - how-to/hibernate-workers.md
```
Comment (enxebre) [inline docs/mkdocs.yml:46]: Let's instead a new section: "Automated Machine Management" and let's this be a subsection of it, e.g. "Scaling down data plane to Zero"

**Comment by enxebre** [comment]
Thanks Juan! Let's please retitle the PR title and desc to remove any reference to hibernation. This is just about how scaling down and draining works.

File: docs/content/how-to/hibernate-workers.md (hunk 1 above)
Comment (jparrill) [reply docs/content/how-to/hibernate-workers.md:2]: Thanks for the feedback :), updating it

**Comment by zanetworker** [comment]
lgtm
